File: app/schemas/weather.py
from datetime import date

from pydantic import BaseModel


class CurrentWeatherResponse(BaseModel):
    city: str
    temp_c: float
    feels_like_c: float
    description: str
    wind_speed: float
    humidity: int | None = None


class ForecastDay(BaseModel):
    date: date
    temp_min_c: float | None = None
    temp_max_c: float | None = None
    wind_speed_avg: float | None = None
    description: str | None = None


class ForecastResponse(BaseModel):
    city: str
    days: list[ForecastDay]

class CitySuggestionResponse(BaseModel):
    name: str
    country: str
    state: str | None = None
    lat: float | None = None
    lon: float | None = None

File: weather-mobile/src/types/weather.ts
export type CurrentWeather = {
  city: string;
  temperature: number;
  description: string;
  feelsLike: number;
  windSpeed: number;
  humidity: number | null;
};

export type ForecastDay = {
  day: string;
  min: number | null;
  max: number | null;
  windSpeed: number | null;
  description: string | null;
};

export type CitySuggestion = {
  name: string;
  country: string;
  state: string | null;
  lat: number | null;
  lon: number | null;
};
